packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/Models/RoundTripModel.Serialization.cs

```diff
@@ -86,8 +86,18 @@ protected virtual void JsonModelWriteCore(Utf8JsonWriter writer, ModelReaderWrit
             writer.WriteEndObject();
             writer.WritePropertyName("requiredModel"u8);
             writer.WriteObjectValue(RequiredModel, options);
-            writer.WritePropertyName("intExtensibleEnum"u8);
-            writer.WriteNumberValue(IntExtensibleEnum.ToSerialInt32());
+            if (Optional.IsDefined(IntExtensibleEnum))
+            {
+                if (IntExtensibleEnum != null)
+                {
+                    writer.WritePropertyName("intExtensibleEnum"u8);
+                    writer.WriteNumberValue(IntExtensibleEnum.Value.ToSerialInt32());
+                }
+                else
+                {
+                    writer.WriteNull("intExtensibleEnum"u8);
+                }
```

Comment:
So this might also need a fix.
The fact that its type is nullable does not mean its serialization is nullable.
Despite it runs the same, we should not generate the same code for
```
optional1?: int32;
optional2?: int32 | null;
```